### PR TITLE
added commands to constructor

### PR DESCRIPTION
**Pull request by OfriRosenbaum**

### CODE REVIEW

File: src/main/java/frc/robot/commands/autonomous/SuperAutonomous.java [hunk 1]
```diff
@@ -18,25 +17,73 @@
...
+                new MoveToCargo(drivetrain, MoveToCargo.CARGO_MOVE_VALUE));
     }
 
-    private DriveTankWithPID returnByEncoders() {
-        return new DriveTankWithPID(drivetrain, drivetrain.getEncodersPIDSettings(), drivetrain.getEncodersPIDSettings(),
-                drivetrain.getLeftDistance() / 2, drivetrain.getRightDistance() / 2,
-                drivetrain::getLeftDistance, drivetrain::getRightDistance);
+    private ParallelRaceGroup returnByHalf() {
+        return new DriveArcade(drivetrain, () -> -MoveToCargo.CARGO_MOVE_VALUE.get(), () -> 0.0).withTimeout(
+                0.5 * (secondCargoFinishTime - secondCargoStartTime));
```
Comment (tal-sitton) [inline src/main/java/frc/robot/commands/autonomous/SuperAutonomous.java:79]: why `* 0.5` instead of `/2`

**Review by tal-sitton — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/MoveToCargoWithIntake.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import frc.robot.subsystems.IntakeToTransfer;
+
+/**
+ * Moves to a cargo, while simultaneously aiming the robot to face the cargo and then intake the cargo.
+ */
+public class MoveToCargoWithIntake extends ParallelCommandGroup {
+
+    public static final double CARGO_MOVE_VALUE = 0.7;
+    public static final double SETPOINT = 10.0;
+    public static final double MOVE_TO_CARGO_TIMEOUT = 4.2212;
```
Comment (tal-sitton) [inline src/main/java/frc/robot/commands/MoveToCargoWithIntake.java]: move this to `GyroAuto`

File: src/main/java/frc/robot/commands/autonomous/SuperAutonomous.java [hunk 3]
```diff
@@ -1,59 +1,107 @@
...
+
+    private ParallelRaceGroup returnByHalf() {
+        return new DriveArcade(drivetrain, () -> -MoveToCargoWithIntake.CARGO_MOVE_VALUE, () -> 0.0).withTimeout(
+                (secondCargoFinishTime - secondCargoStartTime) / 2);
     }
 
     private Command seekHub() {
         Limelight limelight = drivetrain.getLimelight();
         return new DriveArcade(drivetrain, () -> 0.0, SEEK_ROTATE_VALUE).withInterrupt(() ->
                 -SEEK_HUB_TOLERANCE.get() <= limelight.getHorizontalOffsetFromTarget() &&
```
Comment (Yoel-Perman-Brilliant) [inline src/main/java/frc/robot/commands/autonomous/SuperAutonomous.java:94]: `Math.abs` is only 8 characters long

File: src/main/java/frc/robot/commands/autonomous/SuperAutonomous.java [hunk 4]
```diff
@@ -1,59 +1,107 @@
...
-     */
-    private boolean hasCargoTarget() {
-        try {
-            RootNamespace imageProcess = new RootNamespace("Image Processing");
-            Namespace contourInfo = imageProcess.addChild("contour 0");
-            return contourInfo.getBoolean("isUpdated");
-        } catch (Exception e) {
-            return false;
-        }
+    //@todo
```
Comment (Yoel-Perman-Brilliant) [inline src/main/java/frc/robot/commands/autonomous/SuperAutonomous.java:103]: you don't say

**Review by Yoel-Perman-Brilliant — CHANGES_REQUESTED** [review]
good

**Review by tal-sitton — CHANGES_REQUESTED** [review]

**Review by yutytuty — APPROVED** [review]

**Review by tal-sitton — APPROVED** [review]

**Pull request merged**